transactions/transactions_service.go

```diff
@@ -740,6 +796,60 @@ func (svc *transactionsService) ConsumeEvent(ctx context.Context, event *events.
 			}).WithError(err).Error("Failed to execute DB transaction")
 			return
 		}
+
+	case "nwc_lnclient_hold_invoice_accepted":
+		lnClientTransaction, ok := event.Properties.(*lnclient.Transaction)
+		if !ok {
+			logger.Logger.WithField("event", event).Error("Failed to cast event properties for hold invoice accepted")
+			return
+		}
+		logger.Logger.WithFields(logrus.Fields{
+			"paymentHash": lnClientTransaction.PaymentHash,
+		}).Info("Processing hold invoice accepted event")
+
+		var dbTransaction db.Transaction
+		err := svc.db.Transaction(func(tx *gorm.DB) error {
+			result := tx.Where("payment_hash = ? AND type = ? AND state = ?", lnClientTransaction.PaymentHash, constants.TRANSACTION_TYPE_INCOMING, constants.TRANSACTION_STATE_PENDING).First(&dbTransaction)
+			if result.Error != nil {
+				if errors.Is(result.Error, gorm.ErrRecordNotFound) {
+					logger.Logger.WithFields(logrus.Fields{
+						"paymentHash": lnClientTransaction.PaymentHash,
+					}).Warn("No corresponding pending incoming transaction found in DB for accepted hold invoice")
+				}
+				logger.Logger.WithFields(logrus.Fields{
+					"paymentHash": lnClientTransaction.PaymentHash,
+				}).WithError(result.Error).Error("Failed to query DB for accepted hold invoice")
+				return result.Error
+			}
+
+			err := tx.Model(&dbTransaction).Update("state", constants.TRANSACTION_STATE_ACCEPTED).Error
+			if err != nil {
+				logger.Logger.WithFields(logrus.Fields{
+					"paymentHash": lnClientTransaction.PaymentHash,
+					"dbTxID":      dbTransaction.ID,
+				}).WithError(err).Error("Failed to update hold invoice state to accepted in DB")
+				return err
+			}
+			dbTransaction.State = constants.TRANSACTION_STATE_ACCEPTED
+
+			logger.Logger.WithFields(logrus.Fields{
```

Comment:
for json logging we currently use `lower_case_separated_by_underscores` for the field names